src/Type/TypeUtils.php

```diff
@@ -111,6 +112,16 @@ public static function getConstantScalars(Type $type): array
 		return self::map(ConstantScalarType::class, $type, false);
 	}
 
+	/**
+	 * Returns whether $type is a MixedType
+	 *
+	 * FIXME: Remove this once TemplateTypes do not inherit their bound anymore
+	 */
+	public static function isMixedType(Type $type): bool
+	{
+		return $type instanceof MixedType && !$type instanceof TemplateType;
```

Comment:
I don't get why this is needed. Why `$type instanceof MixedType` isn't sufficient? I think it should work the same way.

Reply:
Also, please remove the `FIXME` comment. I'd rather track this in a proper todolist.

Reply:
A `array<K, V>` is described() as `array` instead of `array<K, V>` because K and V, who are represented by TemplateMixedType instances, are also MixedTypes (ArrayType::describe() decides to omit the key type and value type when they are mixed).

Reply:
This helper is useful to check that a type is just a mixed type. I would prefer if TemplateMixedType did not inherit MixedType, but this will require some big refactoring.

Reply:
But TemplateMixedType should be treated as MixedType in 99% of places. I think it's a little misleading to put `TypeUtils::isMixedType` there - people might be tempted to start using it without exactly knowing why. But currently it's useful only for `ArrayType::describe()`. So it should probably be just an inlined condition in that method.